BrainPortal/app/models/boutiques_portal_task.rb

```diff
@@ -640,4 +643,10 @@ def file_access_symbol
     @_file_access ||= (self.class.properties[:readonly_input_files].present? || self.tool_config.try(:inputs_readonly) ? :read : :write)
   end
 
+  private
+
+  def single_file_input?()
+    @single_file_input ||= self.descriptor_for_form.inputs.select{|x| x["type"] == "File"}.size == 1
```

Comment:
1. why do you declare the method with `()` on the `def` line ?
2. You cannot cache a true/false value with a `||=` because in the case of false, you will recompute the expression every time.
3. Use the methods for the inputs! `x.type` instead of `x["type"]` 
4. You use select() to create a temporary array of all the inputs of type File, and then all you do is check if the array has size `1`. Then the array is discarded. That's a waste of memory. Arrays have a count() method that you can pass a block to:
``ruby
desc.inputs.count { |x| x.type == 'File' } == 1
```